docs/design/free_credit_offer_typesDesignDoc.md

```diff
@@ -0,0 +1,253 @@
+  The Default Offer will remain constant, with no expiration date to the offer and does not have a redemption cap
+  The Offer Program Admin is able to adjust the default offer
+  One Default Offer per Tardigrade Satellite.
+  When there is a lapse of the Referral Testing Offer/the Referral Testing Offer has expired or reached a cap, the Default Offer will be the default to the Referral Testing Offer that new users will redeem until a new Referral Testing Offer is generated.
+
+**Referral Testing Offer**:
+  - This is the credit awarded to the Invitee upon creating their account using a Referral Link. This credit can have a different expiration date than the awarded Referrer Credit and the Default Free Credit.
+  An admin of the Referral Program is able to easily adjust the current offer through the  Admin Satellite GUI
+
+**Referrer Award Credit**:
+  - This is the credit awarded to the Referrer once the Invitee has created a *paid* account and has successfully paid their second invoice. This credit can have a different expiration date than the awarded Invitee Credit and the Default Free Credit.
+
+[Offer Types Table](https://docs.google.com/spreadsheets/d/1I3Do-HMNkpUpJAsebtl1NXw6-PlkoVaFKCP9e-TTgHA/edit?usp=sharing)
+
+## Design
+
+### Database
+
+**offer  table**
+```sql
+    Id - int
+    Name -  text
+    Description - text  
+    Credits - integer
+    redeemable_cap - integer
+    Num_redeemed - integer
+    Created_at - timestamp
+    offer_duration_days - int
+    award_credit_duration_days - int
+    Invitee_credit_duration_days - int
+    type - enum[FREE_TIER, REFERRAL]
+    status - enum[ON_GOING, DEFAULT, EXPIRED, NO_STATUS]
+    PRIMARY KEY (id)
+```
+
+**user_credit table**
+```sql
+    User_id - bytea
+    offer_id - int
+    credits_earned - float
+    Credit_type - enum[AWARD, INVITEE, NO_TYPE]
+    expires_at - timestamp
+    Created_at - timestamp
+    referred_by - bytea (nullable)
+    FOREIGN KEY (offer_id)
+    PRIMARY KEY (user_id)
+```
+
+**user table**
+```sql
+    total_Referred - int
+```
+
+### Offer Program Service
+
+**satellite/offer/offer.go**
+- Create offers interface to interact with offer table
+
+```golang
+type Offers interface {
+  ListAllOffers(ctx context.Context) ([]Offer, error)
+  GetOfferById(ctx context.Context, offerId []byte) (Offer, error)
+  Update(ctx context.Context, offer *Offer) (*Offer, error)
+  Delete(ctx context.Context, offerId []byte)
+  Create(ctx context.Context, offer *Offer)
+}
+```
+
+**satellite/console/credit.go**
+- Create a user_credit interface to interact with the user_credit table
+
+```golang
+type Credits interface {
+  AvailableCredits(ctx context.Context, userId uuid.UUID) (int, error)
+  ListByCreditType(ctx context.Context, userId uuid.UUID, creditType Credit.Type) ([]Credit, error)
+  Update(ctx context.Context, credit *Credit) (*Credit, error)
+  Create(ctx context.Context, credit *Credit) (*Credit, error)
+}
+```
+
+**satellite/console/database.go**
```

Comment:
Usually, it isn't necessary to write these things out in this detail, these things aren't too complicated to figure out without the design doc. But, since it's here we can leave it for now.